sympy/polys/matrices/linsolve.py

```diff
@@ -73,6 +74,13 @@ def _linsolve(eqs, syms):
     Aaug = sympy_dict_to_dm(eqsdict, rhs, syms)
     K = Aaug.domain
 
+    # sdm_irref has issues with float matrices. This uses the Matrix.rref()
+    # method. When DomainMatrix.rref() can handle float matrices reasonably
+    # this should be removed...
+    if K.is_RealField:
```

Comment:
Complex field should treated similarly.

Reply:
Yes, it should although it isn't actually used here since it always goes to EX:
```python
In [2]: construct_domain([1.0+1.0*I])
Out[2]: (EX, [EX(1.0 + 1.0*I)])

In [3]: Poly(1.0+1.0*I, x)
Out[3]: Poly(1.0 + 1.0*I, x, domain='EX')
```
I think the only way to get CC is by unifying `ZZ_I/QQ_I` with `RR`:
```python
In [4]: ZZ_I.unify(RR)
Out[4]: CC
```

Reply:
I've changed it so that `construct_domain` can identify `CC`. I would have expected to see some test failures from that but there don't seem to be any which means that there is not good test coverage for Poly/linsolve etc over CC.

It should be possible to test `if K.is_Exact` here but that apparently fails for a polynomial ring:
```python
In [1]: RR.frac_field(x).is_Exact
Out[1]: False

In [2]: RR[x].is_Exact
Out[2]: True
```